### ISSUE TITLE
Implement LFU Cache

### ISSUE THREAD

**Issue by rasmus-beck** [rasmus-beck/FerroJson#2]
The schemas loaded should be cached in a Least Frequently Used cache, so we don't need to parse and create the schema every time we need it.

**Comment by TheNewQAGuy** [comment]
Makes sense. We should probably just cache the parseresult.

I guess there is a number of things that need sorting out as for instance how to determine block size (or schema size) and how to prioritize new entries that have just entered the cache and thus may be at the start of their use cycle over older ones that may have been used a lot previously but are now in fact obsolete. Maybe we can introduce a degrading mechanism for those.

**Comment by rasmus-beck** [comment]
True, LFU might not be the optimal solution, but it comes close I think.
